### ISSUE TITLE
GatsbyImageSharpResolutions GraphQLError: Expected positive integer for height but received 53.5 of type number,GraphQLError:

### ISSUE THREAD

**Issue by eric-personal** [gatsbyjs/gatsby#8638]
## Description
GraphQl error when adding width to the query. 

```
The GraphQL query from /Users/eric.murray/projects/clearme-client/src/pages/partners.js failed.

Errors:
  GraphQLError: Expected positive integer for height but received 53.5 of type number,GraphQLError: Expected positive integer for height but received 70.5 of type number
```

### Steps to reproduce

Using Prismic gatsby-source-prismic 1.2.0 plugin. When I add a image (example size could be any size) 300 x 100 then try to set smaller or larger size (in my graphQl query) than the size set in Prismic custom type I will get a build and develop warning which breaks the build. See error above.

In prismic I have set my custom types to 300. Then in my query I want to control the resolutions width to be 120 or any smaller or larger number. If I match the number the integer error goes away.  

Query sample of localFile

```
        localFile {
          childImageSharp {
            resolutions(quality: 80, width: 120) {
              ...GatsbyImageSharpResolutions_withWebp_tracedSVG
            }
          }
        }
```

### Expected result

Should not see this error. 

### Actual result

```
The GraphQL query from /Users/eric.murray/projects/clearme-client/src/pages/partners.js failed.

Errors:
  GraphQLError: Expected positive integer for height but received 53.5 of type number,GraphQLError: Expected positive integer for height but received 70.5 of type number
```

Breaks build.

### Environment

  System:
    OS: macOS High Sierra 10.13.6
    CPU: x64 Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.11.3 - ~/.nvm/versions/node/v8.11.3/bin/node
    Yarn: 1.7.0 - ~/.nvm/versions/node/v8.11.3/bin/yarn
    npm: 5.6.0 - ~/.nvm/versions/node/v8.11.3/bin/npm
  Browsers:
    Chrome: 69.0.3497.100
    Safari: 12.0
  npmPackages:
    gatsby: ^1.9.247 => 1.9.273 
    gatsby-image: ^1.0.54 => 1.0.54 
    gatsby-link: ^1.6.40 => 1.6.45 
    gatsby-plugin-favicon: ^2.1.1 => 2.1.1 
    gatsby-plugin-google-analytics: ^1.0.31 => 1.0.31 
    gatsby-plugin-google-tagmanager: ^1.0.19 => 1.0.19 
    gatsby-plugin-postcss-sass: ^1.0.21 => 1.0.22 
    gatsby-plugin-react-helmet: ^2.0.10 => 2.0.11 
    gatsby-plugin-react-next: ^1.0.11 => 1.0.11 
    gatsby-plugin-resolve-src: ^1.1.3 => 1.1.3 
    gatsby-plugin-sass: ^1.0.26 => 1.0.26 
    gatsby-plugin-sharp: ^1.6.48 => 1.6.48 
    gatsby-plugin-sitemap: ^1.2.25 => 1.2.25 
    gatsby-remark-images: ^1.5.66 => 1.5.67 
    gatsby-source-filesystem: ^1.5.39 => 1.5.39 
    gatsby-source-prismic: ^1.2.0 => 1.2.0 
    gatsby-transformer-sharp: ^1.6.27 => 1.6.27 
  npmGlobalPackages:
    gatsby-cli: 1.1.58

### Notes
I saw this other ticket https://github.com/gatsbyjs/gatsby/issues/4156 but it looks like that was an older Gatsby version than I'm using and looks like it was fixed in my version. 

Upgrading to Gatsby V2 right now is not an option for this project. Looking for a solution in V1 if possible.

**Comment by kakadiadarpan** [comment]
@eric-personal can you provide a reproduction repo for this? That would make it much easier to diagnose this issue.

**Comment by eric-personal** [comment]
Yes but It's going to take me some time to break it out into its own environment.

**Comment by EgorCloud** [comment]
Same problem: 
```Error: Expected positive integer for left but received 1125.435257915621 of type number```
This is solved with the help of ```Number.parseInt()``` but it removes decimal from Number.
